### ISSUE TITLE
CUDA matmul examples missing CUDA.@sync

### ISSUE THREAD

**Issue by stillyslalom** [ENCCS/julia-for-hpc#56]
CUDA's fast, but it's not so fast that a 2^9 x 2^9 matmul takes the same time as a 2^13 x 2^13 as indicated in the tutorial. For accurate timing, you need to wait for the GPU kernel call to complete. For benchmarking, CUDA calls can be wrapped in `CUDA.@sync` (see the [docs](https://cuda.juliagpu.org/stable/development/profiling/#Time-measurements)).
https://github.com/ENCCS/Julia-for-HPC/blob/1aafd827a6be4637364f5fb098a493f1149a1e1a/content/GPU.rst?plain=1#L285-L327

On my Ryzen 4900/RTX2060-equipped laptop, CUDA still gives a decent speed-up, but certainly not 216,000x!
```
julia> A = rand(Float32, 2^12, 2^12);

julia> A_d = cu(A);

julia> @btime $A * $A;
  313.955 ms (2 allocations: 64.00 MiB)

julia> @btime $A_d * $A_d;
  23.100 μs (28 allocations: 576 bytes)

julia> @btime CUDA.@sync $A_d * $A_d;
  21.891 ms (79 allocations: 3.98 KiB)

**Comment by wikfeldt** [comment]
Many thanks for raising this! Indeed the numbers don't make sense. Will fix this asap

**Comment by stillyslalom** [comment]
Thanks! Great tutorials, by the way - thanks for putting the material together.

**Comment by wikfeldt** [comment]
solved in [ce12057](https://github.com/ENCCS/julia-for-hpc/commit/ce12057d8322026c73c0e57320cb90e680c6b86a)
